Rakefile

```diff
@@ -2,11 +2,12 @@
 
 require "bundler/gem_tasks"
 require "rubygems/package_task"
+require "rake_compiler_dock"
 require "rake/testtask"
 require "rake/extensiontask"
 require "rb_sys"
 
-cross_rubies = %w[3.4.0 3.3.5 3.2.0 3.1.0]
+ruby_cc_version = RakeCompilerDock.ruby_cc_version([">= 3.1", "< 3.5"])
```

Comment:
rake-compiler-dock now needs exact patch versions, so this method abstracts those details.